src/api/internal/sensors/usecase/service.go

```diff
@@ -30,36 +25,16 @@ func NewSensorService(repo repository.SensorRepository) SensorService {
 	return &SensorServiceImpl{Repo: repo}
 }
 
-// Checks the required fields of the Sensor.
-func validateRquiredFields(sensor *domain.Sensor) error {
-	if sensor.Name == "" || sensor.Category == "" {
-		return errors.New("name and category are required")
-	}
-	return nil
-}
+func (s *SensorServiceImpl) ListSensors(ctx context.Context, userUuid uuid.UUID, search string) ([]domain.Sensor, error) {
 
-func (s *SensorServiceImpl) CreateSensor(sensor *domain.Sensor) error {
-	if err := validateRquiredFields(sensor); err != nil {
-		return err
+	var sensors, err = s.Repo.ListSensors(ctx, userUuid, search)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve users: %w", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to retrieve sensors: %w", err)
```
![1632944004487](https://github.com/user-attachments/assets/6bc4c444-a5e3-4a28-bdb8-669353ae5457)